tools/cldr-rdf/src/main/java/org/unicode/cldr/tool/GenerateLanguageContainment.java

```diff
@@ -49,37 +52,78 @@
 public class GenerateLanguageContainment {
     private static final boolean ONLY_LIVING = false;
     private static final CLDRConfig CONFIG = CLDRConfig.getInstance();
+    private static final QueryClient queryClient = QueryClient.getInstance();
+    
     static final Splitter TAB = Splitter.on('\t').trimResults();
     static final CLDRFile ENGLISH = CONFIG.getEnglish();
     static final String relDir = "../util/data/languages/";
     static final Map<String, R2<List<String>, String>> ALIAS_MAP = CONFIG
         .getSupplementalDataInfo()
         .getLocaleAliasInfo()
         .get("language");
-    static final Map<String, String> entityToLabel = loadTsvPairsUnique(GenerateLanguageContainment.class, relDir + "entityToLabel.tsv",
-        null, null, null);
-
+    
+    /**
+     * We load the SparQL queries using this helper object, to be able to catch exceptions…
+     */
+    final static class QueryHelper {
+        final public Map<String, String> entityToLabel;
+        final public Map<String, String> entityToCode;
+        final public ImmutableMultimap<String, String> codeToEntity;
+        final public Multimap<String, String> childToParent;
+        
+        QueryHelper() {
+            try {
+                entityToLabel = loadQueryPairsUnique(GenerateLanguageContainment.class, "wikidata-entityToLabel",
+                null, null, null);
+                
+                entityToCode = loadQueryPairsUnique(GenerateLanguageContainment.class,  "wikidata-entityToCode",
+                        code -> {
+                            code = code.replace("\"", "");
+                            R2<List<String>, String> v = ALIAS_MAP.get(code);
+                            String result = v == null
+                                ? code : v.get0().get(0);
+                            result = result.contains("_")
+                                ? code
+                                : result;
+                            return result;
+                        },
+                        null, NAME);
+                
+                codeToEntity = ImmutableMultimap.copyOf(
+                        Multimaps.invertFrom(Multimaps.forMap(entityToCode), LinkedHashMultimap.create()));
+                
+                childToParent = loadQueryPairs(GenerateLanguageContainment.class, "wikidata-childToParent",
+                        code -> getEntityName(code), code -> getEntityName(code));
+                
+            } catch(Throwable t) {
+                t.printStackTrace();
+                throw new RuntimeException(t);
+            }
+        }
+        
+        String getEntityName(String key) {
+            String code = entityToCode.get(key);
+            if (code != null) {
+                try {
+                    String name = NAME.apply(code);
+                    if (name != null) {
+                        return name;
+                    }
+                } catch (Exception e) {
```

Comment:
why would NAME.apply throw an exception? ideally prevent the exception, or print a warning when it happens

Reply:
I'm not sure— this was existing code from https://github.com/unicode-org/cldr/pull/900/files/89fe206ccd3bc8cfbd4b308148d21e4d190ed14f#diff-f5160be6c0e53da3045f35879f2bab04b51ce622ab6899c90062a6cc1bb3bf44L439 

this calls into a pretty deep stack of code in CLDRFile, it definitely could throw an exception of some sort. 

But for the purpose of this operation, not being able to get a name at that level means that it will fall back to the rest of this function which will try other mechanisms for extracting a name.

This code already needs other work (see #666), the purpose of this PR is to reduce the manual steps  needed to 'turn the crank' on the process.

i'm inclined to not change this code until after the other issues in #666 are addressed.

Reply:
Perhaps I could add a TODO here?